homeassistant/components/update/__init__.py

```diff
@@ -0,0 +1,360 @@
+        SERVICE_INSTALL,
+        {
+            vol.Optional(ATTR_VERSION): cv.string,
+            vol.Optional(ATTR_BACKUP): cv.boolean,
+        },
+        async_install,
+    )
+
+    component.async_register_entity_service(
+        SERVICE_SKIP,
+        {},
+        UpdateEntity.async_skip.__name__,
+    )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up a config entry."""
+    component: EntityComponent = hass.data[DOMAIN]
+    return await component.async_setup_entry(entry)
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    component: EntityComponent = hass.data[DOMAIN]
+    return await component.async_unload_entry(entry)
+
+
+async def async_install(entity: UpdateEntity, service_call: ServiceCall) -> None:
+    """Service call wrapper to validate the call."""
+    # This entity does not support installing updates.
+    if not entity.supported_features & SUPPORT_INSTALL:
+        raise HomeAssistantError(
+            f"Installing updates is not supported for {entity.name}"
+        )
+
+    # If version is not specified, but no update is available.
+    if (version := service_call.data.get(ATTR_VERSION)) is None and (
+        entity.current_version == entity.latest_version or entity.latest_version is None
+    ):
+        raise HomeAssistantError(f"No update available for {entity.name}")
+
+    # If version is specified, but not supported by the entity.
+    if version is not None and not entity.supported_features & SUPPORT_SPECIFIC_VERSION:
+        raise HomeAssistantError(
+            f"Installing a specific version is not supported for {entity.name}"
+        )
+
+    # If backup is requested, but not supported by the entity.
+    if (
+        backup := service_call.data.get(ATTR_BACKUP)
+    ) and not entity.supported_features & SUPPORT_BACKUP:
+        raise HomeAssistantError(f"Backup is not supported for {entity.name}")
+
+    # Update is already in progress.
+    if entity.in_progress is not False:
+        raise HomeAssistantError(
+            f"Update installation already in progress for {entity.name}"
+        )
+
+    await entity.async_install_with_progress(version, backup)
+
+
+@dataclass
+class UpdateEntityDescription(EntityDescription):
+    """A class that describes update entities."""
+
+    device_class: UpdateDeviceClass | str | None = None
+    entity_category: EntityCategory | None = EntityCategory.CONFIG
+
+
+class UpdateEntity(RestoreEntity):
+    """Representation of an update entity."""
+
+    entity_description: UpdateEntityDescription
+    _attr_current_version: str | None = None
+    _attr_device_class: UpdateDeviceClass | str | None
+    _attr_in_progress: bool | int = False
+    _attr_latest_version: str | None = None
+    _attr_release_summary: str | None = None
+    _attr_release_url: str | None = None
+    _attr_state: None = None
```

Comment:
`_attr_state` is not used?

Reply:
`_attr_state` is inherited from `Entity`. 
This is merely here to override the typing, preventing it from being set (e.g, if an integration sets `_attr_state = "on"`, mypy will trigger, as the type doesn't match.
We use this trick on more entity components.